### ISSUE TITLE
Extmarks get moved without any apparent change to the line.

### ISSUE THREAD

**Issue by L3MON4D3** [hrsh7th/nvim-compe#381]
When expanding `ccls` snippets they expand just fine with correct extmark-placement. Directly after, however, extmarks are moved to both ends of the line. This issue only appears after ~7d46b1c~ 888d9ec and on a very recent neovim version (`NVIM v0.5.0-dev+1329-gc46d98146`) and, so far, I have only noticed it on `ccls` (`rust-analyzer` works fine).
This causes issues with eg. LuaSnip, where extmarks are used to mark node-positions.

#### Reproduce
0. Install `ccls` and latest `neovim`
1. `nvim -u` [minconf.log](https://github.com/hrsh7th/nvim-compe/files/6504210/minconf.log)
2. edit cpp-file, expand snippet
3. call `:lua Luasnip_current_nodes[vim.api.nvim_get_current_buf()].parent:dump()` to see moved extmarks.

#### Actual behavior
Extmarks are preserved on the line.
#### Expected behavior
Extmarks are moved.
#### Screenshots (optional)
![screen](https://user-images.githubusercontent.com/41961280/118717909-f0041700-b826-11eb-8280-b79398d38c5e.png)
First four are directly after expansion, second four were dumped manually.

**Comment by hrsh7th** [comment]
Please detect why rust-analyzer works fine but ccls isn't.

It is too hard to investigate at the moment.

**Comment by L3MON4D3** [comment]
I tested some more stuff and found out that the issue also exists with `rust-analyzer`, I just didn't notice it before because I only tested the `match`-postfix
```
match server {
    ${1:_} => {$0},
}
```
and there are no marks in the first line that could be displaced :|
I tested the texlab-lsp and there not the first line is affected, but the last.
#### Rust-Analyzer
From lsp.log for 
```
       server.|
``` 
where `|` is the cursor.
```lua
"decoded" {
  id = 2,
  jsonrpc = "2.0",
  result = {
    isIncomplete = true,
    items = {
      ...
      {
        additionalTextEdits = {
          {
            newText = "",
            range = {
              end = {
                character = 15,
                line = 10
              },
              start = {
                character = 8,
                line = 10
              }
            }
          }
        },
        deprecated = false,
        detail = "match expr {}",
        filterText = "match",
        insertTextFormat = 2,
        kind = 15,
        label = "match",
        textEdit = {
          newText = "match server {\n    ${1:_} => {$0},\n}",
          range = {
            end = {
              character = 17,
              line = 10
            },
            start = {
              character = 15,
              line = 10
            }
          }
        }
      },
      ...
  }
}
```
The issue also persists for items where `additionalTextEdits = {}`.
##### Texlab
```tex
    \|
```
```lua
"decoded"	{
  id = 5,
  jsonrpc = "2.0",
  result = {
    data = "commandSnippet",
    detail = "built-in",
    insertText = "begin{$1}\n\t$0\n\\end{$1}",
    insertTextFormat = 2,
    kind = 15,
    label = "begin",
    preselect = false,
    sortText = "00"
  }
}
```
I can't really make out what the changes in 7d46b1c cause and how they could be related to this issue.

**Comment by L3MON4D3** [comment]
I was mistaken, the issue was introduced in 888d9ec, not 7d46b1c. Terribly sorry, I hope I didn't cause any unnecessary headaches because of this :(

**Comment by hrsh7th** [comment]
I think it was fixed by f9151af
